cni/pkg/taint/taint.go

```diff
@@ -0,0 +1,387 @@
+package taint
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	client "k8s.io/client-go/kubernetes"
+	"k8s.io/kubectl/pkg/util/podutils"
+
+	"istio.io/pkg/log"
+)
+
+const (
+	TaintName     = "NodeReadiness"
+	ToleranceName = "NodeReadiness"
+)
+
+type ConfigSettings struct {
+	name           string
+	namespace      string
+	LabelSelectors string `json:"label_selectors"`
+}
+
+func (config ConfigSettings) ToString() string {
+	return fmt.Sprintf("name is %s, namespace is %s,  selector is %s", config.name, config.namespace, config.LabelSelectors)
+}
+
+type Options struct {
+	ConfigName      string
+	ConfigNameSpace string
+}
+type TaintSetter struct {
+	configs []ConfigSettings // contains all configmaps information
+	Client  client.Interface
+}
+
+func (ts TaintSetter) GetAllConfigs() []ConfigSettings {
+	return ts.configs
+}
+func NewTaintSetter(clientset client.Interface, options *Options) (ts TaintSetter, err error) {
+	configmap, err := clientset.CoreV1().ConfigMaps(options.ConfigNameSpace).Get(context.TODO(), options.ConfigName, metav1.GetOptions{})
+	if err != nil {
+		return ts, err
+	}
+	ts = TaintSetter{
+		configs: []ConfigSettings{},
+		Client:  clientset,
+	}
+	ts.LoadConfig(*configmap)
+	return
+}
+
+// load corresponding configmap's critical labels and their namespace
+func (ts *TaintSetter) LoadConfig(config v1.ConfigMap) {
+	log.Debugf("Loading configmap %s in %s", config.Name, config.Namespace)
+	ts.configs = make([]ConfigSettings, 0) //clear previous one
+	for key, value := range config.Data {
+		log.Debugf("Loading %s ", key)
+		temp := strings.Split(value, "\n")
+		var tempset ConfigSettings
+		for _, str := range temp {
+			if strings.HasPrefix(strings.TrimSpace(str), "name=") {
```

Comment:
You should use a library to parse these, not hand parsing. Strongly prefer switching to yaml as well